lib/src/react_client/private_utils.dart

```diff
@@ -0,0 +1,231 @@
+        "cannot be skipped and will still be added to the new component. Please "
+        "remove them from skipMethods.");
+  }
+
+  return finalList;
+}
+
+/// Converts a list of variadic children arguments to children that should be passed to ReactJS.
+///
+/// Returns:
+///
+/// - `null` if there are no args and [shouldAlwaysBeList] is false
+/// - `[]` if there are no args and [shouldAlwaysBeList] is true
+/// - the single child if only one was specified
+/// - otherwise, the same list of args, will all top-level children validated
+dynamic generateChildren(List childrenArgs, {bool shouldAlwaysBeList = false}) {
+  var children;
+
+  if (childrenArgs.isEmpty) {
+    if (!shouldAlwaysBeList) return null;
+    children = childrenArgs;
+  } else if (childrenArgs.length == 1) {
+    if (shouldAlwaysBeList) {
+      final singleChild = react_client_utils.listifyChildren(childrenArgs.single);
+      if (singleChild is List) {
+        children = singleChild;
+      }
+    } else {
+      children = childrenArgs.single;
+    }
+  }
+
+  if (children is Iterable && children is! List) {
+    children = children.toList(growable: false);
+  }
+
+  if (children == null) {
+    children = shouldAlwaysBeList ? childrenArgs.map(react_client_utils.listifyChildren).toList() : childrenArgs;
+    markChildrenValidated(children);
+  }
+
+  return children;
+}
+
+/// Converts [props] into a [JsMap] that can be utilized with `React.createElement()`.
+JsMap generateJsProps(Map props,
+    {bool shouldConvertEventHandlers = true,
+    bool convertRefValue = true,
+    bool convertCallbackRefValue = true,
+    bool wrapWithJsify = true}) {
+  final propsForJs = JsBackedMap.from(props);
+
+  if (shouldConvertEventHandlers) convertEventHandlers(propsForJs);
+  if (convertRefValue) convertRefValue2(propsForJs, convertCallbackRefValue: convertCallbackRefValue);
+
+  return wrapWithJsify ? jsifyAndAllowInterop(propsForJs) : propsForJs.jsObject;
+}
+
+@Deprecated('6.0.0')
+InteropContextValue jsifyContext(Map<String, dynamic> context) {
+  var interopContext = new InteropContextValue();
+  context.forEach((key, value) {
+    // ignore: argument_type_not_assignable
+    setProperty(interopContext, key, new ReactDartContextInternal(value));
+  });
+
+  return interopContext;
+}
+
+/// A wrapper around [validateJsApi] that will return the result of a callback if React is present.
+T validateJsApiThenReturn<T>(T Function() computeReturn) {
+  validateJsApi();
+  return computeReturn();
+}
+
+@Deprecated('6.0.0')
+Map<String, dynamic> unjsifyContext(InteropContextValue interopContext) {
+  // TODO consider using `contextKeys` for this if perf of objectKeys is bad.
+  return new Map.fromIterable(_objectKeys(interopContext), value: (key) {
+    // ignore: argument_type_not_assignable
+    ReactDartContextInternal internal = getProperty(interopContext, key);
+    return internal?.value;
+  });
+}
+
+/// A function that verifies React can can be accessed.
```

Comment:
This could be a little more descriptive. Also #nit function doc comments [should start with verbs](https://dart.dev/guides/language/effective-dart/documentation#prefer-starting-function-or-method-comments-with-third-person-verbs) according to the Dart style guide.
```dart
/// Validates that React JS has been loaded and its APIs made available on the window,
/// throwing an exception otherwise.
void validateJsApi() {
```